004-governance.Rmd

```diff
@@ -0,0 +1,153 @@
+
+## Solution
+
+The following sections describes a governance model to use as a default for open source RStudio repositories (starting with those in the r-lib, tidyverse, and tidymodels organisations).
+It is not mandatory, but it has been designed to reflect our current best practices, and should be used unless there are compelling reasons to favour a different approach.
+
+The model strikes a balance between the [benevolent dictator](http://oss-watch.ac.uk/resources/benevolentdictatorgovernancemodel) and [rough consensus](https://datatracker.ietf.org/doc/html/rfc7282) governance models.
+We blend these two governance models because we want to involve the community in decision making as much as possible, while recognising that excellent user experience often requires consistent design guided by a single voice.
+
+### Users
+
+People who use the package are the most important members of the community; without these users, this project would have no purpose.
+Users are encouraged to participate in the life of the project and the community as much as possible.
+Common user activities include (but are not limited to):
+
+-   Evangelising about the project.
+-   Asking and answering questions on community forums.
+-   Providing moral support (a "thank you" goes a long way).
+
+### Contributors
+
+Users who continue to engage with the project and its community will often find themselves becoming more and more involved.
+Such users may then go on to become **contributors** by interacting with the project on GitHub.
+Contributors:
+
+-   Report bugs and suggest improvements by creating new issues.
+
+-   Improving existing issues by answering questions, creating reprexes, or providing feedback on proposed changes.
+
+-   Contributing code or documentation via pull requests.
+
+Anyone can become a contributor: there is no expectation of commitment to the project, no required set of skills, and no selection process.
+The only requirement is to follow the code of conduct.
+
+Packages don't maintain an explicit list of contributors but acknowledge them in blog posts using data from GitHub aggregated by `usethis::use_tidy_thanks()`.
+
+### Authors
+
+Contributors who have made significant and sustained contributions can be invited to become authors.
+Authors are collectively responsible for day-to-day development of the package, including responding to issues and reviewing pull requests.
+An author possesses two special powers:
+
+-   They have **write** access on GitHub so they can triage issues, request review on PRs, and merge them.
+
+-   They are listed in `Authors@R` so they receive credit when others cite the package.
+
+Authors are expected to follow our standard processes, such as:
+
+-   **Code contribution**: code is usually contributed via PR, even for authors who could push directly.
+    Particularly high-stakes project may want to protect the main branch and require "request reviews before merging".
```

Comment:
Has anyone actually done this? I wonder if it would make normal development frustratingly slow for small stuff. But maybe maintainer has a permission level that allows them to commit straight to a protected branch.

Reply:
Yes I have done it on a number of repos. Maintainers can still commit to the main branch, and merge PRs without extra approval.

Reply:
Ah, ok then that removes my hesitance.